src/main/groovy/au/org/ala/ecodata/metadata/OutputUploadTemplateBuilder.groovy

```diff
@@ -437,7 +437,7 @@ class ValidationHandler implements OutputModelProcessor.Processor<ExcelValidatio
 
     @Override
     def text(Object node, ExcelValidationContext context) {
-        if (node.constraints) {
+        if (node.constraints && node.constraints instanceof org.grails.web.json.JSONObject == false) {
```

Comment:
Introducing this condition will break the download of computed constraints with options (the next if statement on line 443).
As an aside, you should be using Map as a test, not JSONObject as the issue is casting a Map to a List, not the specific implementation of a Map.

Reply:
Thanks for pointing me that out, I just saw that the ESP SMU Reporting has a computed type constraints.
I've updated the validation for the pre-populated constraints type.